framework/Ramsh/include/Ramsh/RamshTypeInfo.h

```diff
@@ -175,6 +175,17 @@ namespace ramses_internal
         }
     };
 
+#ifdef __APPLE__
```

Comment:
why is it different than the existing template for Int64? Does it really resolve ti Int64?

Reply:
The reason is that clang enforces that `long` and `long long` are different types, so the `Int64` and `Int` instantiations are different, even though they might have the same bitness on most platform. If we check in `PlatformTypes.h` and the standard library how the types resolve:

```
ramses_internal::Int --> std::intptr_t --> long
ramses_internal::Int32 --> int32_t --> int
ramses_internal::Int64 --> int64_t --> long long
```

I get an compile error on macOS if I do not have the `Int` version present. The question is, is the `#ifdef`even necessary? Should I add a comment explaning the situation in the code?

Reply:
It looks like the Int template specialization is not present currently, so I'd say remove the ifdef (provided it works). Doesn't look like iOS specific issue, I just wonder why it only occurs on AppleClang (we already have a clang build - there it works). Maybe that's one of the subtle apple-specific differences?

Reply:
Ok, I removed the #ifdef. Yeah I think that is one of the differences.

Reply:
I undid the change again, since it does not work: https://github.com/bmwcarit/ramses/actions/runs/3901946222/jobs/6664404935

Apparently, `int64_t` resolves to `long` on Ubuntu and `long long` on macOS.